### ISSUE TITLE
Support for remote info stores like Confluence, Sharepoint etc possible?

### ISSUE THREAD

**Issue by emanueol** [arc53/docsgpt#27]
Or must all files exist locally ?

In real world of large enterprises, theres a confluence server, a jira server, sharepoint server that typically reside in a data center or as a SaaS cloud, and some on-prem custom html, excel files etc.

Would be great a ChatGPT type of ingest/compute searches across remote systems.. how feasible is this ?
Thanks

**Comment by dartpain** [comment]
Thats a good feature we can start working on, as loang as we can prep it in a neat and readable format we can ingest it all. But I do think we have to vectorise all this data first or summarise at least.

**Comment by terrafying** [comment]
I was able to pull confluence data by exporting the whole space as HTML, changing the glob pattern in ingest_rst.py to point to the downloaded files, and using BeautifulSoup to pull out the text content inside the main <div> tags.

**Comment by emanueol** [comment]
I suppose theres 2 things:

> initial export (Confluence allows exporting a space in couple different formats).

> incremental updates (Jira supports webhooks , sending json via http post to some listener)

> Or maybe easier to prep a clone so you gpt stuff doesn't interfer with users, but it all boils down to file formats.

> im not specialist of under the hood of Confluence, Jira, etc.. but im part of those 0.001% that care centralizing knowledge for both tech guys (majority) and satisfy business/stakeholders. 
Its a very interesting point of discussion "hiw to decide organization of information, but i suppose some basic high level stuff could be inserted on the source pages (tags, etc) to help with categorization, as least avoid showing code to business manager when he just looking for the 5 business rules agreed with Development in some project. etc.

**Comment by dartpain** [comment]
I think what needs to be built here is just a module for our parser. basically something that loads data and converts it into (.rst, .md, .pdf, .docx, .csv, .epub, .html) As DocsGPT already loads this files with easy. Its just method of scraping files that we need to implement here

**Comment by tardigrde** [comment]
FYI: This nice repo implemented loading text from YT videos as well as websites
https://github.com/embedchain/embedchain

**Comment by KennyDizi** [comment]
Deeplake could be the good fit?
https://github.com/activeloopai/deeplake

**Comment by dartpain** [comment]
Also @pabik is already working on it in the feature/remote-loads branch
We will also need to do the UI in parallel

**Comment by thefoodiecoder** [comment]
Llama Index provides support for ingestion from these sources. We can either look into integrating or porting these. 

- **Confluence**: https://llamahub.ai/l/readers/llama-index-readers-confluence?from=
- **Notion**: https://llamahub.ai/l/readers/llama-index-readers-notion?from=all
- **Google Docs & Sheets**: https://llamahub.ai/l/readers/llama-index-readers-google?from=all

> All loaders - https://llamahub.ai/?tab=readers

**Comment by thefoodiecoder** [comment]
Python isn't my forte but let me see if someone from my Data Science team is willing to contribute.

**Comment by adsdemaybe** [comment]
Is there a reason why this issue is still open?

**Comment by dartpain** [comment]
Im just waiting to merge the Sharepoint

**Comment by dartpain** [comment]
for Confluence I created a separate issue, closing this one 
Confluence issue:
#2360
